### ISSUE TITLE
Elasticsearch ARM build does not include license public.key

### ISSUE THREAD

**Issue by pebrc** [elastic/elasticsearch#68083]
When trying to [update the Elasticsearch license](https://www.elastic.co/guide/en/elasticsearch/reference/current/update-license.html) on ARM e.g. with `PUT _license` Elasticsearch responds with HTTP 500 and the following stack trace: 
```
"stacktrace": ["java.lang.NullPointerException: Cannot invoke \"java.io.InputStream.read(byte[])\" because \"in\" is null",
"at org.elasticsearch.core.internal.io.Streams.copy(Streams.java:52) ~[elasticsearch-core-7.10.1.jar:7.10.1]",
"at org.elasticsearch.core.internal.io.Streams.copy(Streams.java:86) ~[elasticsearch-core-7.10.1.jar:7.10.1]",
"at org.elasticsearch.license.LicenseVerifier.verifyLicense(LicenseVerifier.java:80) ~[?:?]",
"at org.elasticsearch.license.LicenseService.registerLicense(LicenseService.java:212) ~[?:?]",
"at org.elasticsearch.license.TransportPutLicenseAction.masterOperation(TransportPutLicenseAction.java:56) ~[?:?]",
"at org.elasticsearch.license.TransportPutLicenseAction.masterOperation(TransportPutLicenseAction.java:25) ~[?:?]",
"at org.elasticsearch.action.support.master.TransportMasterNodeAction.masterOperation(TransportMasterNodeAction.java:99) ~[elasticsearch-7.10.1.jar:7.10.1]",
"at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$3(TransportMasterNodeAction.java:166) ~[elasticsearch-7.10.1.jar:7.10.1]",
"at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73) ~[elasticsearch-7.10.1.jar:7.10.1]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737) [elasticsearch-7.10.1.jar:7.10.1]",
"at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.10.1.jar:7.10.1]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]",
"at java.lang.Thread.run(Thread.java:832) [?:?]"] }
```

which seems to indicate that Elasticsearch is failing to load its public key to verify the license

https://github.com/elastic/elasticsearch/blob/ba3ceeaccf9b59ad16af60e279dbd6339e091a48/x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseVerifier.java#L78-L80

A comparison of the x-pack-core jars on amd64 and arm64 via `docker cp` shows that the public key file seems missing on ARM: 

```sh
~ jar tf x-pack-core-7.10.1.jar | grep key
public.key
~ jar tf x-pack-core-7.10.1.arm.jar | grep key
~
```

This was discovered here https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-eks-arm/3/console

**Comment by pugnascotia** [comment]
* I checked in `https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.10.2-linux-aarch64.tar.gz` and `public.key` exists in `modules/x-pack-core/x-pack-core-7.10.2.jar`.
* I ran the Docker image `docker.elastic.co/elasticsearch/elasticsearch:7.10.1` on an `x86_64` host, and the key file is present in the jar
* I ran the same image on an ARM host, but this time the key file is missing
* I built a Docker image on an ARM host with the following, and the key file was present:
   ```
   ./gradlew -Dbuild.snapshot=false -Dlicense.key=$PWD/unused.key distribution:docker:buildAarch64DockerImage
   ```

So it appears that something about how we're building the ARM Docker image in the release manager is broken.

**Comment by pugnascotia** [comment]
We fixed the release manager build.

**Comment by lucabelluccini** [comment]
Comment for anyone who might hit this problem.

Elasticsearch docker images for ARM have been fixed in 7.11.1 and more recent versions.

Still, it is worth mentioning Elasticsearch is officially supported on ARM only on 7.12.0+ as per https://www.elastic.co/support/matrix#matrix_os
